app/bundles/EmailBundle/Tests/Controller/Api/EmailApiControllerFunctionalTest.php

```diff
@@ -203,4 +219,232 @@ public function testReplyAction(): void
+        $clientResponse  = $this->client->getResponse();
+
+        $this->assertSame(201, $clientResponse->getStatusCode(), $clientResponse->getContent());
+
+        $segmentResponse = json_decode($clientResponse->getContent(), true);
+        $segmentId       = $segmentResponse['list']['id'];
+
+        $this->assertGreaterThan(0, $segmentId, $clientResponse->getContent());
+
+        // Add contact to segment:
+        $this->client->request('POST', "/api/segments/${segmentId}/contact/${contactId}/add");
+        $clientResponse  = $this->client->getResponse();
+
+        $this->assertSame(200, $clientResponse->getStatusCode(), $clientResponse->getContent());
+
+        // Create an email:
+        $payload = [
+            'name'           => 'API email',
+            'subject'        => 'Email created via API test',
+            'emailType'      => 'list',
+            'lists'          => [$segmentId],
+            'customHtml'     => '<h1>Email content created by an API test</h1>',
+            'isPublished'    => 1,
+            'fromAddress'    => 'from@api.test',
+            'fromName'       => 'API Test',
+            'replyToAddress' => 'reply@api.test',
+            'bccAddress'     => 'bcc@api.test',
+        ];
+
+        $this->client->request('POST', '/api/emails/new', $payload);
+        $clientResponse = $this->client->getResponse();
+
+        $this->assertSame(201, $clientResponse->getStatusCode(), $clientResponse->getContent());
+
+        $emailResponse  = json_decode($clientResponse->getContent(), true);
+        $emailId        = $emailResponse['email']['id'];
+
+        $this->assertGreaterThan(0, $emailId, $clientResponse->getContent());
+
+        // Send to segment:
+        $this->client->request('POST', "/api/emails/${emailId}/send");
+        $clientResponse = $this->client->getResponse();
+        $sendResponse   = json_decode($clientResponse->getContent(), true);
+
+        $this->assertSame(200, $clientResponse->getStatusCode(), $clientResponse->getContent());
+        $this->assertEquals($sendResponse, ['success' => true, 'sentCount' => 1, 'failedRecipients' => 0], $clientResponse->getContent());
+
+        $testEmail = function (): void {
+            $message = $this->transport->sentMessage;
+            $this->assertSame($message->getSubject(), 'Email created via API test');
+            $bodyRegExp = '#<h1>Email content created by an API test</h1><img height="1" width="1" src="[^"]+" alt="" />#';
+            $this->assertMatchesRegularExpression($bodyRegExp, $message->getBody());
+            $this->assertSame($message->getTo(), ['jane@api.test' => 'Jane Doe']);
+            $this->assertSame($message->getFrom(), ['from@api.test' => 'API Test']);
+            $this->assertSame($message->getReplyTo(), ['reply@api.test' => null]);
+            $this->assertSame($message->getBcc(), ['bcc@api.test' => null]);
+        };
+        $testEmail();
```

Comment:
I'd like to learn more about this technique. Is there any other reason for this named anonymous function other than naming what is going on inside? What is the advantage over creating a private function in the test class?

Reply:
The anonymous function here is simply a reusable block of code, since I needed to perform the same verifications twice, here
https://github.com/mautic/mautic/blob/85eb220c346d11944715172174237f7eca932cd9/app/bundles/EmailBundle/Tests/Controller/Api/EmailApiControllerFunctionalTest.php#L331
and here
https://github.com/mautic/mautic/blob/85eb220c346d11944715172174237f7eca932cd9/app/bundles/EmailBundle/Tests/Controller/Api/EmailApiControllerFunctionalTest.php#L342
I preferred to not repeat myself.